### ISSUE TITLE
Google testing fails on icon_map_generator Icon "Too few positional arguments: 2 required, 1 given."

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#101611]
Seen in cl/440438444 during unrelated https://github.com/flutter/flutter/pull/100787
```
google3:///[mobile/flutter/tools/icon_font/bin/icon_map_generator.dart:46](https://cs.corp.google.com/piper///depot/google3/mobile/flutter/tools/icon_font/bin/icon_map_generator.dart?l=46&ws=tap-presubmit-rerun-server/627796&snapshot=2):22: Error: Too few positional arguments: 2 required, 1 given.
    final icon = Icon(entry);
                     ^
google3:///[third_party/dart/flutter/tools/lib/update_icons.dart:393](https://cs.corp.google.com/piper///depot/google3/third_party/dart/flutter/tools/lib/update_icons.dart?l=393&ws=tap-presubmit-rerun-server/627796&snapshot=2):3: Context: Found this candidate, but the arguments don't match.
  Icon(MapEntry<String, String> tokenPair, this.fontFamily) {
  ^^^^


ERROR: mobile/flutter/tools/icon_font/bin/icon_map_generator.dart

46: final icon = Icon(entry); 2 positional argument(s) expected, but 1 found. #not_enough_positional_arguments
```
Caused by https://github.com/flutter/flutter/pull/101190, needs a g3fix?
@guidezpl @clocksmith

**Comment by guidezpl** [comment]
This was confusing to me because the fix is already checked in. I accidentally "pre-patched it" with cl/439647158, see http://google3/third_party/dart/flutter/tools/lib/update_icons.dart;l=471;rcl=440193506. But, Google testing isn't using the code at HEAD, for some unknown reason, which is why this fails.

**Comment by jmagman** [comment]
I don't think patching third_party/dart/flutter outside of a roll would be correct. I think you should instead have made a [G3Fix](https://g3doc.corp.google.com/company/teams/core/developer/acx/flutter/process/life_of_a_pr/index.md?cl=head#g3fix) with just the `mobile/flutter/tools/icon_font/bin/icon_map_generator.dart` change and apply it in FRoB to https://github.com/flutter/flutter/pull/100787

**Comment by jmagman** [comment]
Changing to P1, this is causing Google Testing to fail on every commit past https://github.com/flutter/flutter/pull/101190.
@guidezpl can you work with infra to get the g3fix?

**Comment by jmagman** [comment]
Adding to infra ticket queue to help navigate the FRoB errors.

**Comment by godofredoc** [comment]
\cc @CaseyHillers @XilaiZhang

**Comment by CaseyHillers** [comment]
The issue is these generated CLs are copying over a change in `third_party/dart/flutter/tools/lib/update_icons.dart` even though the presubmit PRs don't contain that file.

**Comment by CaseyHillers** [comment]
Once the current roll lands (ETA ~1 hour), we should see the breakage stop.

I'm investigating why this occurred in the first place. I'm checking to see if the CL head is wrong, or if the roll script is erroneously copying this file over.

**Comment by guidezpl** [comment]
I'm out for a few days but just wanted to confirm what happened while I was working on a PR and CL at the same time.

On April 5th, https://github.com/flutter/flutter/pull/101190#event-6370168332 was merged, and a little over 8 hours later, cl/439647158 was submitted. While `update_icons.dart` is expected to be rolled in, I noticed that the CL accidentally submitted the change to `update_icons.dart` that would be expected to be rolled in later (due to working on two separate things touching the same file). 

I was surprised to see that a few days later, generated CLs were then testing using an old HEAD. Adding a g3fix didn't make that much sense, since the "fix" was already in at HEAD.
